keel-core/src/main/kotlin/com/netflix/spinnaker/keel/services/ResourceStatusService.kt

```diff
@@ -103,15 +103,46 @@ class ResourceStatusService(
   }
 
   /**
-   * Checks last 10 events for flapping between only ResourceActuationLaunched and ResourceDeltaDetected
+   * Returns true if a resource has been vetoed by the unhappy veto,
+   * or if the last 10 events are only ResourceActuationLaunched or ResourceDeltaDetected events,
+   * or if the resource has been vetoed by an unspecified veto that we don't have an explicit status mapping for.
```

Comment:
Oof! This is a bit involved. But I don't have a great suggestion to simplify at the moment... 😕

Reply:
This got simplified a little, which is nice